### PR TITLE
Improve Web Links doc

### PR DESCRIPTION
**Pull request by ia3andy**
- Improve description
- Add guide links
- Links is already included in rest data panache (no need to have it in the doc dependencies)

### CODE REVIEW

**Comment by ia3andy** [comment]
I can't find information about the annotation for Web Links with Resteasy Classic

File: extensions/resteasy-reactive/quarkus-resteasy-reactive-links/runtime/pom.xml [hunk 1]
```diff
@@ -11,7 +11,7 @@
 
     <artifactId>quarkus-resteasy-reactive-links</artifactId>
     <name>Quarkus - RESTEasy Reactive - Links - Runtime</name>
-    <description>Quarkus RESTEast Reactive resource linking support</description>
+    <description>Quarkus RESTEast Reactive resource linking support (Web Links). This module will allow injecting web links into the response HTTP headers by just annotating your endpoint resources.</description>
```
Comment (gsmet) [inline extensions/resteasy-reactive/quarkus-resteasy-reactive-links/runtime/pom.xml:14]:
```suggestion
    <description>Quarkus RESTEasy Reactive resource linking support (Web Links). This module will allow injecting web links into the response HTTP headers by annotating your endpoint resources.</description>
```

**Review by gsmet — CHANGES_REQUESTED** [review]

File: docs/src/main/asciidoc/rest-data-panache.adoc [hunk 2]
```diff
@@ -107,15 +107,11 @@ implementation("io.quarkus:quarkus-resteasy-reactive-jackson")
...
-   <!-- Use these if you are using RESTEasy Reactive Jackson -->
+   <!-- Use this if you are using RESTEasy Reactive Jackson for serialization -->
     <dependency>
         <groupId>io.quarkus</groupId>
         <artifactId>quarkus-resteasy-reactive-jackson</artifactId>
     </dependency>
-    <dependency>
-        <groupId>io.quarkus</groupId>
-        <artifactId>quarkus-resteasy-reactive-links</artifactId>
-    </dependency>
```
[the remark this answers is not in the document]
Comment (ia3andy) [reply docs/src/main/asciidoc/rest-data-panache.adoc:118]: It's already a dependency of rest data panache (and reactive)

[the remark this answers is not in the document]
Comment (ia3andy) [reply docs/src/main/asciidoc/rest-data-panache.adoc:118]: https://github.com/quarkusio/quarkus/blob/main/extensions/panache/hibernate-reactive-rest-data-panache/runtime/pom.xml#L31

[the remark this answers is not in the document]
Comment (ia3andy) [reply docs/src/main/asciidoc/rest-data-panache.adoc:118]: https://github.com/quarkusio/quarkus/blob/main/extensions/panache/rest-data-panache/runtime/pom.xml#L39

**Review by gsmet — APPROVED** [review]

**Pull request merged**